homeassistant/components/wilight/switch.py

```diff
@@ -0,0 +1,327 @@
+
+        if self.watering_time is not None:
+            attr[ATTR_WATERING_TIME] = self.watering_time
+
+        if self.trigger_1 is not None:
+            attr[ATTR_TRIGGER_1] = self.trigger_1
+
+        if self.trigger_2 is not None:
+            attr[ATTR_TRIGGER_2] = self.trigger_2
+
+        if self.trigger_3 is not None:
+            attr[ATTR_TRIGGER_3] = self.trigger_3
+
+        if self.trigger_4 is not None:
+            attr[ATTR_TRIGGER_4] = self.trigger_4
+
+        if self.trigger_1_description is not None:
+            attr[ATTR_TRIGGER_1_DESC] = self.trigger_1_description
+
+        if self.trigger_2_description is not None:
+            attr[ATTR_TRIGGER_2_DESC] = self.trigger_2_description
+
+        if self.trigger_3_description is not None:
+            attr[ATTR_TRIGGER_3_DESC] = self.trigger_3_description
+
+        if self.trigger_4_description is not None:
+            attr[ATTR_TRIGGER_4_DESC] = self.trigger_4_description
+
+        return attr
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        return ICON_WATERING
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the device on."""
+        await self._client.turn_on(self._index)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the device off."""
+        await self._client.turn_off(self._index)
+
+    async def async_set_watering_time(self, watering_time: int) -> None:
+        """Set the watering time."""
+        await self._client.set_switch_time(self._index, watering_time)
+
+    async def async_set_trigger(self, trigger_index: int, trigger: str) -> None:
+        """Set the trigger according to index."""
+        if trigger_index == 1:
+            await self._client.set_switch_trigger_1(self._index, trigger)
+        if trigger_index == 2:
+            await self._client.set_switch_trigger_2(self._index, trigger)
+        if trigger_index == 3:
+            await self._client.set_switch_trigger_3(self._index, trigger)
+        if trigger_index == 4:
+            await self._client.set_switch_trigger_4(self._index, trigger)
+
+
+class WiLightValvePauseSwitch(WiLightDevice, SwitchEntity):
+    """Representation of a WiLights Valve Pause switch."""
+
+    @property
+    def name(self) -> str:
+        """Return the name of the switch."""
+        return f"{self._attr_name} {DESC_PAUSE}"
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on."""
+        return self._status.get("on", False)
+
+    @property
+    def pause_time(self) -> int | None:
+        """Return pause time of valve switch.
+
+        None is unknown, 1 is minimum, 24 is maximum.
+        """
+        pause_time = self._status.get("timer_target")
+        if pause_time is not None:
+            return wilight_to_hass_pause_time(pause_time)
+        return pause_time
+
+    @property
+    def state_attributes(self) -> dict[str, Any]:
```

Comment:
Please go over the code and check that my comments have been applied to all of the entities.
```suggestion
    def extra_state_attributes(self) -> dict[str, Any]:
```

Reply:
Done